packages/@sanity/base/src/preview/createPreviewObserver.ts

```diff
@@ -1,17 +1,20 @@
 import {of as observableOf, Observable} from 'rxjs'
-import {map, switchMap} from 'rxjs/operators'
+import {map, switchMap, catchError} from 'rxjs/operators'
 import {isReferenceSchemaType, ReferenceSchemaType, SchemaType} from '@sanity/types'
-import prepareForPreview, {invokePrepare, PrepareInvocationResult} from './prepareForPreview'
-import {FieldName, Path, Reference, PrepareViewOptions} from './types'
+import prepareForPreview, {invokePrepare, PreparedValue} from './prepareForPreview'
+import {Reference, PrepareViewOptions} from './types'
+import {INSUFFICIENT_PERMISSIONS_FALLBACK, InsufficientPermissionsError} from './constants'
 
-export interface PreviewValue {
+const INSUFFICIENT_PERMISSIONS = Symbol('INSUFFICIENT_PERMISSIONS')
+
+export interface PreparedSnapshot {
   type?: SchemaType
-  snapshot: null | PrepareInvocationResult
+  snapshot: null | PreparedValue
 }
 
 // Takes a value and its type and prepares a snapshot for it that can be passed to a preview component
 export function createPreviewObserver(
-  observePaths: (value: any, paths: Path[]) => any,
```

Comment:
Why was this changed?

Reply:
I think this was unintentional. I will revert.